services/cd-service/pkg/service/git_test.go

```diff
@@ -38,134 +41,429 @@ import (
 		givenEnv               *string
 		givenEnvGroup          *string
-		expectedProductSummary []api.ProductSummary
-		expectedErr            error
-		Setup                  []rp.Transformer
+		expectedProductSummary []*api.GetProductSummaryResponse
+		expectedErrors         []StageError
+		SetupStages            [][]rp.Transformer
 	}{
-		{
-			Name:        "get Product Overview with no env or envGroup",
-			expectedErr: fmt.Errorf("Must have an environment or environmentGroup to get the product summary for"),
-		},
-		{
-			Name:        "get Product Overview with no commitHash",
-			givenEnv:    conversion.FromString("testing"),
-			expectedErr: fmt.Errorf("Must have a commit to get the product summary for"),
-		},
-		{
-			Name:          "get Product Overview with both env and envGroup",
-			givenEnv:      conversion.FromString("testing"),
-			givenEnvGroup: conversion.FromString("testingGroup"),
-			expectedErr:   fmt.Errorf("Can not have both an environment and environmentGroup to get the product summary for"),
-		},
 		{
 			Name:     "get Product Overview as expected with env",
 			givenEnv: conversion.FromString("development"),
-			Setup: []rp.Transformer{
-				&rp.CreateEnvironment{
-					Environment: "development",
-					Config: config.EnvironmentConfig{
-						Upstream: &config.EnvironmentConfigUpstream{
-							Latest: true,
+			SetupStages: [][]rp.Transformer{
+				{
+					&rp.CreateEnvironment{
+						Environment: "development",
+						Config: config.EnvironmentConfig{
+							Upstream: &config.EnvironmentConfigUpstream{
+								Latest: true,
+							},
+							ArgoCd:           nil,
+							EnvironmentGroup: conversion.FromString("dev"),
 						},
-						ArgoCd:           nil,
-						EnvironmentGroup: conversion.FromString("dev"),
 					},
-				},
-				&rp.CreateApplicationVersion{
-					Application: "test",
-					Manifests: map[string]string{
-						"development": "dev",
+					&rp.CreateApplicationVersion{
+						Application: "test",
+						Manifests: map[string]string{
+							"development": "dev",
+						},
+						SourceAuthor:    "example <example@example.com>",
+						SourceCommitId:  "deadbeefdeadbeefdeadbeefdeadbeefdeadbeef",
+						SourceMessage:   "changed something (#678)",
+						DisplayVersion:  "v1.0.2",
+						Team:            "sre-team",
+						WriteCommitData: true,
+						Version:         1,
+					},
+					&rp.DeployApplicationVersion{
+						Application: "test",
+						Environment: "development",
+						Version:     1,
 					},
-					SourceAuthor:    "example <example@example.com>",
-					SourceCommitId:  "testing25",
-					SourceMessage:   "changed something (#678)",
-					DisplayVersion:  "v1.0.2",
-					Team:            "sre-team",
-					WriteCommitData: true,
 				},
-				&rp.DeployApplicationVersion{
-					Application: "test",
-					Environment: "development",
-					Version:     1,
+			},
+			expectedProductSummary: []*api.GetProductSummaryResponse{
+				{
+					ProductSummary: []*api.ProductSummary{
+						{
```

Comment:
how can this test work, if the commitId is not set?

Reply:
What does this mean? The manifest commit hash?😅

Reply:
I mixed this up.